File: Java/102.java
import java.util.*;

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null) {
            return new ArrayList<>();
        }
        List<List<Integer>> result = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while (!queue.isEmpty()) {
            int limit = queue.size();
            List<Integer> curLevel = new ArrayList<>();
            for (int i = 0; i < limit; i++) {
                TreeNode polled = queue.poll();
                curLevel.add(polled.val);
                if (polled.left != null) {
                    queue.offer(polled.left);
                }
                if (polled.right != null) {
                    queue.offer(polled.right);
                }
            }
            result.add(curLevel);
        }
        return result;
    }
}

File: Python/102.py
from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        if not root:
            return result
        queue = []
        queue.append(root)
        while queue:
            cur_level = []
            length_of_queue = len(queue)
            for i in range(length_of_queue):
                popped = queue.pop(0)
                if popped.left:
                    queue.append(popped.left)
                if popped.right:
                    queue.append(popped.right)

                cur_level.append(popped.val)
            result.append(cur_level)

        return result
